core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -461,6 +462,7 @@ public synchronized boolean isReady() {
 
       @Override
       public void cancel(Status reason) {
+        reason = stripCause(reason);
```

Comment:
`reason` is passed two places: one should include the cause, the other shouldn't. I think (and I may be wrong) `internalCancel()` should use `reason` unmodified and serverStream.clientCancelled should be something like `Status.CANCELLED.withDescription("Client cancelled")`.

Does that agree with what you were thinking?